apps/gitness/src/pages-v2/project/labels/hooks/use-get-project-label-and-values-data.ts

```diff
@@ -123,13 +128,19 @@ export const useGetProjectLabelAndValuesData = ({
    * Set space_ref to store
    */
   useEffect(() => {
-    setRepoSpaceRef({
-      space_ref: space_ref ?? ''
-    })
+    setRepoSpaceRef({ space_ref: space_ref ?? '' })
   }, [space_ref, setRepoSpaceRef])
 
-  return {
-    isLoading: isLoadingSpaceLabels || isLoadingValues,
-    space_ref
-  }
+  /**
+   * Set loading state to store
+   */
+  useEffect(() => {
+    if (enabled) {
+      return setIsLoading(isLoadingSpaceLabels)
+    }
+
+    setIsLoading(isLoadingValues)
```

Comment:
It _probably_ isn't an issue, but it's typically a good idea to only return cleanup functions within a `useEffect`. It can be pretty easily rewritten in a few different ways, but personally I'd consider a ternary
```suggestion
    setIsLoading(enabled ? isLoadingSpaceLabels : isLoadingValues)
```